tests/unit/crossing_symmetry_test.py

```diff
@@ -34,20 +33,27 @@ def three_particle_integral_test(params):
     sterile = Particle(**NuP.dirac_sterile_neutrino(mass=200 * UNITS.MeV))
     neutral_pion = Particle(**SMP.hadrons.neutral_pion)
 
-    pion_interactions = NuI.sterile_pion_neutral(theta=1e-3, sterile=sterile,
-                                                 active=neutrino_e, pion=neutral_pion)
+    theta = 1e-3
+    thetas = defaultdict(float, {
+        'electron': theta,
+    })
+
+    interaction = NuI.sterile_hadrons_interactions(
+        thetas = thetas, sterile=sterile,
+        neutrinos = [neutrino_e],
+        leptons = [],
+        mesons = [neutral_pion]
+    )
 
     universe = Universe(params=params)
     universe.add_particles([photon, neutrino_e, sterile, neutral_pion])
-    universe.interactions += pion_interactions
+    universe.interactions += interaction
 
     params.update(universe.total_energy_density(), universe.total_entropy())
-
-    print(universe.interactions)
     assert len(universe.interactions) == 2
-    print(universe.interactions[0].integrals)
-    print(universe.interactions[1].integrals)
     assert len(universe.interactions[0].integrals) == 2
-    assert len(universe.interactions[1].integrals) == 3
+    assert len(universe.interactions[1].integrals) == 2
```

Comment:
Probably it's time to put a comment here explaining why 2 and not 3 and what we are counting in the end.

Reply:
Sterile.interaction.with.hadrons creates something like:

N + nu --> pi
pi + nu --> N

and

nu + N --> pi
pi + anti-nu --> anti-N 

So, there are two of them and each one consists of two objects.

Reply:
You are showing me the crossing processes of a single reaction. Why then there are 2 items in interactions? Shouldn't there be another reaction?

Reply:
Code in NuMSM looks like

[CrossGeneratingInteraction(something) for antiparticles in [
            ((False, ), (False, False)),
            ((True, ), (True, False))
        ]

Where should the third one come from?

Reply:
Now you are showing me `interactions`. There are 2 items because you put inside 2 items. But the test counts integrals inside each interactions. Previously I expected there to be 3 integrals inside one of the interactions. I'm worried that now you put 2 instead of 3 just such that the test passes and not because it is correct. I want to understand why 2 integrals

Reply:
I don't understand what your worry is based on. I used your part of the code to generate the interactions. You never told me what is exactly wrong with the output of CrossGeneratingInteraction in this case. 

The code generates:

N --> nu_e + pi^0
pi^0 + anti-nu_e --> anti-N

pi^0 + nu_e --> N
nu_e + pi^0 --> N

Doing it manually I get:

N --> nu_e + pi^0
N + anti-nu_e --> pi^0

nu_e + anti-N --> pi^0
nu_e + pi^0 --> N

pi^0 --> anti-nu_e + N
pi^0 --> nu_e + anti-N
pi^0 + nu_e --> N
pi^0 + anti-nu_e --> anti-N

Clearly, the function doesn't return the correct reactions. Where do your 3 integrals come from?

Reply:
If mass HNL < mass pion, there should be integrals for the reactions:

N + anti-nu_e <--> pion
nu_e + anti-N <--> pion
pi^0 <--> anti-nu_e + N
pi^0 <--> nu_e + anti-N

If mass HNL > mass pion, there should be integrals for the reactions:

N <--> nu_e + pi^0
nu_e + pi^0 <--> N
pi^0 +nu_e <--> N
pi^0 + anti-nu_e <--> anti-N

The code does this correctly.

In both cases there should be 2 x 2 integrals generated by CrossGeneratingInteraction, which indeed is what my test is checking.